app/uk/gov/hmrc/gform/upscan/UpscanService.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.upscan
+
+import akka.util.ByteString
+import cats.implicits._
+import java.time.Instant
+import play.api.libs.json.Json
+import scala.concurrent.{ ExecutionContext, Future }
+import uk.gov.hmrc.crypto.{ Crypted, CryptoWithKeysFromConfig, PlainText }
+import uk.gov.hmrc.gform.config.AppConfig
+import uk.gov.hmrc.gform.gformbackend.GformConnector
+import uk.gov.hmrc.gform.sharedmodel.AccessCode
+import uk.gov.hmrc.gform.sharedmodel.form.{ EnvelopeId, Form, FormIdData, UserData }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ FormComponentId, FormTemplateId, SectionNumber }
+import uk.gov.hmrc.gform.upscan.routes.UpscanController
+import uk.gov.hmrc.http.HeaderCarrier
+
+class UpscanService(
+  upscanConnector: UpscanConnector,
+  upscanRepository: UpscanRepository,
+  gformConnector: GformConnector,
+  queryParameterCrypto: CryptoWithKeysFromConfig,
+  appConfig: AppConfig
+)(implicit ec: ExecutionContext)
+    extends UpscanAlgebra[Future] {
+
+  def upscanInitiate(
+    fileUploadIds: List[FormComponentId],
```

Comment:
why is this a collection, I was under the assumption its always going to be a single file per page?